crates/primitives/src/transaction/mod.rs

```diff
@@ -1816,6 +1816,32 @@ impl IntoRecoveredTransaction for TransactionSignedEcRecovered {
     }
 }
 
+impl TryFrom<reth_rpc_types::Transaction> for TransactionSignedEcRecovered {
+    type Error = ConversionError;
+
+    fn try_from(tx: reth_rpc_types::Transaction) -> Result<Self, Self::Error> {
+        tx.signature.ok_or(ConversionError::MissingSignature).and_then(|signature| {
+            TransactionSigned::from_transaction_and_signature(
+                tx.try_into()?,
+                Signature {
+                    r: signature.r,
+                    s: signature.s,
+                    odd_y_parity: signature
+                        .y_parity
+                        .ok_or(ConversionError::SignatureError(
+                            alloy_primitives::SignatureError::InvalidParity(0),
```

Comment:
To be fixed by https://github.com/alloy-rs/core/pull/607

Reply:
Please make a separate type or add it to ConversionError

Reply:
Just did it here: https://github.com/alloy-rs/alloy/pull/586